helm-chart/templates/analytics/ingress.yaml

```
@@ -1,48 +1,38 @@
 {{- $version := .Capabilities.KubeVersion.GitVersion -}}
 {{- if and .Values.ingress.enabled .Values.analytics.enabled -}}
 {{- $fullName := include "cvat.fullname" . -}}
-{{- if semverCompare ">=1.19-0" $version -}}
 apiVersion: networking.k8s.io/v1
-{{- else if semverCompare ">=1.14-0" $version -}}
-apiVersion: networking.k8s.io/v1beta1
-{{- else -}}
-apiVersion: extensions/v1beta1
-{{- end }}
 kind: Ingress
 metadata:
   name: {{ $fullName }}-analytics
   namespace: {{ .Release.Namespace }}
   labels:
     {{- include "cvat.labels" . | nindent 4 }}
   annotations:
+    {{- if .Values.traefik.enabled }}
     traefik.ingress.kubernetes.io/router.middlewares: {{ $.Release.Namespace }}-stripprefix@kubernetescrd,
       {{ $.Release.Namespace }}-forwardauth@kubernetescrd
-    {{- with .Values.analytics.ingress.annotations }}
+    {{- end }}
+    {{- with .Values.ingress.annotations }}
       {{- toYaml . | nindent 4 }}
     {{- end }}
 spec:
-{{- if semverCompare ">=1.19-0" $version }}
-  ingressClassName: {{ .Values.ingress.className }}
+{{- if .Values.ingress.ingressClassName }}
+  ingressClassName: {{ .Values.ingress.ingressClassName | quote }}
```

```suggestion
{{- with .Values.ingress.className }}
  className: {{ toYaml . }}
```

Note: the change from `ingress.className` to `ingress.ingressClassName` seems gratuitous, so I'm suggesting to revert it.